### PR TITLE
Contrast 38919 add creating application support

### PR DESCRIPTION
**Pull request by bryanateoan**
> Warning: Do not merge, ask Bryan to fix sdk version before merging.
> Depends on: https://github.com/Contrast-Security-OSS/contrast-sdk-java/pull/34

# Problem 
Jenkins plugin does not implement the new create application methods

# Solution
Implement and make use of new methods

# Summary of Changes
1. Added a boolean flag (createAppIfNotExist) to control if an application should be created if an application is not found.
1. Added required fields (agentType) and made applicationName a required field if createAppIfNotExist is true
1. Added optional fields for applicationCode and applicationPath

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -96,7 +96,8 @@
     <dependency>
       <groupId>com.contrastsecurity</groupId>
       <artifactId>contrast-sdk-java</artifactId>
-      <version>2.9</version>
+    <!--  <version>2.9</version> TODO: CHANGE ME -->
```
Comment (bryanateoan) [inline pom.xml:99]: Tell Bryan to fix before merging

File: src/main/java/com/aspectsecurity/contrast/contrastjenkins/VulnerabilityTrendStep.java [hunk 2]
```diff
@@ -165,17 +203,22 @@ public Step newInstance(Map<String, Object> arguments) {
...
+                if (applicationId != null && !step.isCreateAppIfNotExist()) {
+                    step.setApplicationId(applicationId);
+                } else if (step.isCreateAppIfNotExist()) {
+                    if (applicationId == null || applicationId.isEmpty()) {
+                        step.setApplicationId(applicationId);
+                    } else {
+                        throw new IllegalArgumentException("If createAppIfNotExist is true, Application ID must not be set");
+                    }
                 } else {
                     throw new IllegalArgumentException("Application ID must be set.");
```
Comment (bryanateoan) [inline src/main/java/com/aspectsecurity/contrast/contrastjenkins/VulnerabilityTrendStep.java:217]: not part of this PR but, are fields that throw this exception supposed to be required? They do not get validated if they are not part of the arguments

**Review by gmoore — APPROVED** [review]

**Comment by bryanateoan** [comment]
need to make some changes, closing for now
